src/main/context-handler.ts

```diff
@@ -58,11 +58,18 @@ export class ContextHandler {
   }
 
   public async getPrometheusDetails(): Promise<PrometheusDetails> {
-    const service = await this.getPrometheusService();
-    const prometheusPath = this.ensurePrometheusPath(service);
-    const provider = this.ensurePrometheusProvider(service);
-
-    return { prometheusPath, provider };
+    if (this.prometheusPath == null) {
+      const service = await this.getPrometheusService();
```

Comment:
It would be better if `getPrometheusService` checked this and then returned the appropriate `PrometheusService`

Reply:
I won't be able to fix it before Monday.